sdcm/tester.py

```diff
@@ -915,8 +921,9 @@ def set_system_auth_rf(self, db_cluster=None):
                              num_retry_on_failure=3)
         self.log.debug("system_auth description: %s", res.stdout)
         self.log.info('repair system_auth keyspace ...')
-        for node in self.db_cluster.nodes:
-            node.run_nodetool(sub_cmd="repair system_auth")
+        parallel_objects = ParallelObject(self.db_cluster.nodes, num_workers=min(
+            32, len(self.db_cluster.nodes)), timeout=HOUR_IN_SEC * 2)
```

Comment:
2 hours for this repair? isn't it too much?

Reply:
without the parallelism it can take more the 2h
regardless it's depends on the number of nodes you setup, at the beginning

so multi-dc usually starts from 9 nodes, scale-tests also can start from a quite high number.

so we should set it high enough

@soyacz maybe we should use the adaptive_timeout ? so we could measure it ? (and make the timeout on the number of nodes ?)

Reply:
1st, they are now running in parallel in all nodes... so, this timeout should be technically for each repair on `system_auth` table... how is it possible it takes hours?? if that is the case, we have a huge issue, since this table should be almost empty

Reply:
I was rather thinking to measure how long it takes and put hard timeout on it (or soft one, but without adaptive timeout as currently all the factors we measure don't impact this repair time. Worth to consider some additional metrics put to ES). I think so because this is one of the first things we do in cluster, no nemesis, no load, no data - should always be fast and deterministic. This hard timeout should take into considerations number of nodes (or anything that affects repair speed).

Reply:
> 1st, they are now running in parallel in all nodes... so, this timeout should be technically for each repair on `system_auth` table... how is it possible it takes hours?? if that is the case, we have a huge issue, since this table should be almost empty

Actually we have issue with that in multi-dc - that's why this PR was created

Reply:
> Actually we have issue with that in multi-dc - that's why this PR was created

before we were running repair in a serial manner, but now they are in parallel... for a tiny table... 2 hours to have all of them, regardless the amount of nodes, sounds too much... but a run or two will prove it

Reply:
Yes, I'll be running tests with this to see actual numbers. Without `-pr` for multi-dc scenario with 9 nodes (together) repair of one node took almost 1h. Currently it takes long because there's plenty of requests per range (not table rows) and in multi-dc network delay greatly impacts it.

Reply:
> > Actually we have issue with that in multi-dc - that's why this PR was created
> 
> before we were running repair in a serial manner, but now they are in parallel... for a tiny table... 2 hours to have all of them, regardless the amount of nodes, sounds too much... but a run or two will prove it

Sounds weird, according to:
https://github.com/scylladb/scylla-enterprise/issues/3103

The numbers of ranges that need to be checked across all the nodes in the RF, has nothing to do with the amount of data.